### PR TITLE
YD-659 Upgrade to Spring Boot 2.2.2

### PR DESCRIPTION
**Pull request by Bert-R**
This includes Spring Hateoas 1.0.2, which implies a significant API change since the 0.25.1 version we were on.

We also could revert the fix for the double percent encoding, as that's now solved in Spring Hateoas.

### CODE REVIEW

File: appservice/src/main/java/nu/yona/server/goals/rest/GoalController.java [hunk 1]
```diff
@@ -165,26 +166,23 @@ private GoalResourceAssembler createResourceAssembler(UUID requestingUserId, UUI
...
 
 	private void setActivityCategoryId(GoalDto goal)
 	{
-		Link activityCategoryLink = goal.getLink(curieProvider.getNamespacedRelFor(ACTIVITY_CATEGORY_REL));
-		if (activityCategoryLink == null)
-		{
-			throw InvalidDataException.missingActivityCategoryLink();
-		}
+		Link activityCategoryLink = goal.getLink(curieProvider.getNamespacedRelFor(ACTIVITY_CATEGORY_REL))
+				.orElseThrow(() -> InvalidDataException.missingActivityCategoryLink());
```
Comment (yonabuild) [inline appservice/src/main/java/nu/yona/server/goals/rest/GoalController.java]: MINOR Replace this lambda with a method reference. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS1612)

File: appservice/src/main/java/nu/yona/server/subscriptions/rest/BuddyController.java [hunk 2]
```diff
@@ -17,13 +17,14 @@
 import java.util.UUID;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
 import org.springframework.hateoas.Link;
```
Comment (yonabuild) [inline appservice/src/main/java/nu/yona/server/subscriptions/rest/BuddyController.java]: MINOR Remove this unused import 'org.springframework.hateoas.Link'. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by yonabuild** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: core/src/main/java/nu/yona/server/properties/SecurityProperties.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
 	private String ovpnProfileFile;
 
 	/**
-	 * If true, Cross Origin Resource Sharing is allowed. This is necessary for Swagger UI.
+	 * If true, Cross Origin EntityModel Sharing is allowed. This is necessary for Swagger UI.
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/properties/SecurityProperties.java]: COEMS? Never heard of...

File: appservice/src/main/java/nu/yona/server/analysis/rest/UserActivityController.java [hunk 4]
```diff
@@ -410,16 +412,18 @@ public DayActivityOverviewWithBuddiesResource(UUID requestingUserId, UUID reques
 
 		public List<DayActivityWithBuddiesResource> getDayActivities()
 		{
-			return new DayActivityWithBuddiesResourceAssembler(requestingUserId, requestingDeviceId, goalIdMapping,
-					getContent().getDateStr()).toResources(getContent().getDayActivities());
+			CollectionModel<DayActivityWithBuddiesResource> collectionModel = new DayActivityWithBuddiesResourceAssembler(
+					requestingUserId, requestingDeviceId, goalIdMapping, getContent().getDateStr())
+							.toCollectionModel(getContent().getDayActivities());
+			return Lists.newArrayList(collectionModel);
```
[the remark this answers is not in the document]
Comment (Bert-R) [reply appservice/src/main/java/nu/yona/server/analysis/rest/UserActivityController.java]: We're using Guava in two more places. This is a common way to convert an ``Iterable`` into a ``List`` for people that use Guava.

File: core/src/main/java/nu/yona/server/properties/SecurityProperties.java (hunk 3 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/properties/SecurityProperties.java]: :) This is a funny one caused by the Spring HATEOAS migration script.
I'll correct it.

File: appservice/src/main/java/nu/yona/server/goals/rest/GoalController.java [hunk 5]
```diff
@@ -238,19 +236,19 @@ private void addActivityCategoryLink(GoalDto goalResource)
...
+		private void addSelfLink(WebMvcLinkBuilder selfLinkBuilder, GoalDto goalResource)
 		{
 			goalResource.add(selfLinkBuilder.withSelfRel());
 		}
 
-		private void addEditLink(ControllerLinkBuilder selfLinkBuilder, GoalDto goalResource)
+		private void addEditLink(WebMvcLinkBuilder selfLinkBuilder, GoalDto goalResource)
 		{
-			goalResource.add(selfLinkBuilder.withRel(JsonRootRelProvider.EDIT_REL));
+			goalResource.add(selfLinkBuilder.withRel(JsonRootLinkRelationProvider.EDIT_REL));
```
[the remark this answers is not in the document]
Comment (Bert-R) [reply appservice/src/main/java/nu/yona/server/goals/rest/GoalController.java]: Yes, that's possible. Done.

File: appservice/src/main/java/nu/yona/server/analysis/rest/ActivityControllerBase.java [hunk 6]
```diff
@@ -75,10 +78,10 @@
 	protected static final int WEEKS_DEFAULT_PAGE_SIZE = 2;
 	protected static final int DAYS_DEFAULT_PAGE_SIZE = 3;
 	protected static final int MESSAGES_DEFAULT_PAGE_SIZE = 4;
-	protected static final String PREV_REL = "prev"; // IANA reserved, so will not be prefixed
```
[the remark this answers is not in the document]
Comment (Bert-R) [reply appservice/src/main/java/nu/yona/server/analysis/rest/ActivityControllerBase.java]: Yes, good one. Done.

File: appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy [hunk 7]
```diff
@@ -438,11 +439,11 @@ class CreateUserOnBuddyRequestTest extends AbstractAppServiceIntegrationTest
 		User richard = addRichard()
 		def mobileNumberBob = makeMobileNumber(timestamp)
 		def responseAddBuddy = sendBuddyRequestForBobby(richard, mobileNumberBob)
-		// Take invite URL and remove "tempPassword=abcd&" or "&tempPassword=abcd" (varying order occurs)
-		def urlToTry = getInviteUrl().replaceFirst(/tempPassword=[^&]*&/, "").replaceFirst(/&tempPassword=[^&]*/, "")
+		// Take invite URL and remove "tempPassword=abcd&" or "&tempPassword=ab&cd" (varying order occurs)
```
[the remark this answers is not in the document]
Comment (Bert-R) [reply appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy]: It wasn't only confusing, it was wrong. Corrected it into:
```
// Take invite URL and remove "tempPassword=ab%26cd&" or "&tempPassword=ab%26cd" (varying order occurs)
```

**Comment by Bert-R** [comment]
Pushed an update

**Pull request merged**